modules/openlmis-web/src/main/webapp/public/js/distribution/controller/visit-info-controller.js

```diff
@@ -106,4 +114,27 @@ function VisitInfoController($scope, distributionService, $routeParams) {
 
     return { original: field.original, type: 'reading' };
   }
+
+  function setApplicableVisitInfoForTechnicalStaff(visit) {
+    if (typeof visit.technicalStaff === 'number') {
+      visit.technicalStaff = {
+        type: "reading",
+        defaultValue: visit.technicalStaff,
+        value: visit.technicalStaff
+      };
+    } else if (!visit.technicalStaff || !visit.technicalStaff.value) {
+      visit.technicalStaff = {
+        original: {
+          type: "reading",
+          value: 0
+        },
+        type: "reading",
+        value: 0
+      };
+    } else if (visit.technicalStaff.defaultValue) {
+      visit.technicalStaff.value = visit.technicalStaff.defaultValue;
+    } else if ($scope.isCurrentPeriod()) {
+      visit.technicalStaff = setApplicableField(visit.technicalStaff);
+    }
+  }
```

Comment:
The purpose of all the above conditionals isn't immediately clear. Would you mind adding comments for them, please, so that developers with a less holistic understanding of the context can more readily understand?